tools/killswitch/src/killswitch.rs

```diff
@@ -0,0 +1,258 @@
+    pub(crate) async fn new(args: Args, settings: Settings) -> Result<Self> {
+        let channels = if args.all {
+            Self::make_channels(&settings)
+        } else {
+            let destination_network = args
+                .all_inbound
+                .expect("Should not happen. Clap requires this to be present");
+            let all = Self::make_channels(&settings);
+            Self::make_inbound_channels(&destination_network, all)
+        };
+        if channels.is_empty() {
+            // The one error we bail on, since there's nothing else left to do
+            return Err(Error::NoNetworks);
+        }
+
+        let futs = channels.into_iter().map(|channel| async {
+            let home_contract = Self::make_home(&channel, &settings).await;
+            let connection_manager = Self::make_connection_manager(&channel, &settings).await;
+            let attestation_signer = Self::make_signer(&channel, &settings).await;
+            ChannelKiller {
+                channel,
+                home_contract: Some(home_contract),
+                connection_manager: Some(connection_manager),
+                attestation_signer: Some(attestation_signer),
+            }
+        });
+        let channel_killers = join_all(futs).await.into_iter().collect::<Vec<_>>();
+
+        Ok(Self { channel_killers })
+    }
+
+    /// Collect all blocking errors, returning a `KillSwitch` with a set of channels
+    /// that can actually fire off transactions as well as any errors collected
+    pub(crate) async fn get_blocking_errors(self) -> (Self, Option<Message>) {
+        let (mut failed, maybe_ok): (Vec<_>, Vec<_>) = self
+            .channel_killers
+            .into_iter()
+            .partition(|killer| killer.has_errors());
+
+        let _blocking_errors = failed
+            .iter_mut()
+            .map(|killer| (killer.channel.clone(), killer.take_all_errors()))
+            .collect::<Vec<(_, _)>>();
+
+        // Produce errors to stream before running txs
+        let message = None;
+
+        (
+            KillSwitch {
+                channel_killers: maybe_ok,
+            },
+            message,
+        )
+    }
+
+    /// Run `KillSwitch` against remaining, non-blocked channels
+    pub(crate) async fn run(mut self) -> Message {
+        let futs = self
+            .channel_killers
+            .iter_mut()
+            .map(|killer| async {
+                let futs = async {
+                    let signed_failure = killer.create_signed_failure().await?;
+                    killer.kill(&signed_failure).await
+                }
+                .await;
+                (killer.channel.clone(), futs)
+            })
+            .collect::<Vec<_>>();
+
+        let results = join_all(futs).await;
+
+        let (mut _failed, mut _success): (Vec<_>, Vec<_>) =
+            results.into_iter().partition(|(_, result)| result.is_err());
+
+        // Build results object
+
+        Message::Results //
```

Comment:
Just flagging this as incomplete in case forgotten. Guessing we'd want to collect results and output in `main`

Reply:
Yep, thank